File: Applications/CSharp/Stack/DailyTemperatures.cs
int[] DailyTemperatures(int[] temperatures)
{
    var stack = new Stack<(int, int)>();
    stack.Push((temperatures[0], 0));
    var result = new int[temperatures.Length];

    for (int i = 1; i < temperatures.Length; i++)
    {
        var t = temperatures[i];
        while (stack.Count > 0 && stack.Peek().Item1 < t)
        {
            var item = stack.Pop();
            result[item.Item2] = i - item.Item2;
        }

        stack.Push((t, i));
    }

    return result;
}

File: Applications/Python/Stack/DailyTemperatures.py
# 739. Daily Temperatures

def dailyTemperatures(self, temperatures):
    stack = []
    result = [0] * len(temperatures)

    for i, t in enumerate(temperatures):
        while stack and t > stack[-1][0]:
            _ , idx = stack.pop()
            result[idx] = i - idx
        stack.append((t, i))

    return result